### PR TITLE
Feature/oasys

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/oasys/model/CtAreaEst.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    private String ctAreaEstCode;
+    @Column(name = "AREA_EST_CTID")
+    private String areaEstCtid;
+    @Column(name = "AREA_EST_NAME")
+    private String areaEstName;
+    @Column(name = "START_DATE")
+    private Time startDate;
+    @Column(name = "END_DATE")
+    private Time endDate;
+    @Column(name = "LAO_CONTACT_NUMBER")
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/digital/hmpps/oauth2server/oasys/model/CtAreaEst.java]: remove all these fields not used by the application

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/oasys/model/OasysUser.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    private Long failedLoginAttempts;
+    @Column(name = "SYSTEM_IND")
+    private String systemInd;
+    @OneToOne
+    @JoinColumns({
+            @JoinColumn(name = "USER_STATUS_CAT", referencedColumnName = "REF_CATEGORY_CODE"),
+            @JoinColumn(name = "USER_STATUS_ELM", referencedColumnName = "REF_ELEMENT_CODE")
+    })
+    private RefElement userStatus;
+    @Column(name = "DATE_OF_BIRTH")
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/digital/hmpps/oauth2server/oasys/model/OasysUser.java]: remove etc

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/oasys/model/RefElement.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+@Table(name = "REF_ELEMENT", schema = "EOR")
+@IdClass(RefElementPK.class)
+public class RefElement {
+    @Column(name = "REF_ELEMENT_UK")
+    private Long refElementUk;
+    @Id
+    @Column(name = "REF_CATEGORY_CODE")
+    private String refCategoryCode;
+    @Id
+    @Column(name = "REF_ELEMENT_CODE")
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/digital/hmpps/oauth2server/oasys/model/RefElement.java]: Remove unneeded

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/oasys/model/RefRole.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    private String refRoleCode;
+    @Column(name = "REF_ROLE_DESC")
+    private String refRoleDesc;
+    @Column(name = "ROLE_TYPE_ELM")
+    private String roleTypeElm;
+    @Column(name = "ROLE_TYPE_CAT")
+    private String roleTypeCat;
+    @Column(name = "CT_AREA_EST_CODE")
+    private String ctAreaEstCode;
+    @Column(name = "RESTRICTED_IND")
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/digital/hmpps/oauth2server/oasys/model/RefRole.java]: Remove unused

File: src/main/resources/db/oasys/V10_1__user_tables.sql [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+ALTER TABLE EOR.OASYS_USER
+    ADD FOREIGN KEY (CT_AREA_EST_CODE)
+        REFERENCES CT_AREA_EST (CT_AREA_EST_CODE);
+
+create table EOR.OFFICE
+(
+    OFFICE_UK         NUMBER        not null
+        constraint OFFICE_UN
+            unique,
+    OFFICE_CODE       VARCHAR2(100) not null,
```
Comment (Mjwillis) [inline src/main/resources/db/oasys/V10_1__user_tables.sql]: Again, suspect these are not needed

File: src/main/resources/db/dev/data/oasys/V10_3__role_data.sql [hunk 6]
```diff
@@ -0,0 +1,39 @@
+INSERT INTO EOR.DIVISION (DIVISION_UK, CT_AREA_EST_CODE, DIVISION_CODE, DIVISION_NAME, START_DATE, END_DATE, CHECKSUM, CREATE_DATE, CREATE_USER, LASTUPD_DATE, LASTUPD_USER) VALUES (151, 'CMB', 'CMB Default LDU', 'CMB Default LDU', TO_DATE('2013-03-20 00:00:00', 'YYYY-MM-DD HH24:MI:SS'), null, 'CD68E7F685EF9D66DB5D494085ECBF9E', TO_DATE('2012-10-16 10:16:25', 'YYYY-MM-DD HH24:MI:SS'), 'EOR_RETAIN_DATA', TO_DATE('2013-03-20 12:23:44', 'YYYY-MM-DD HH24:MI:SS'), 'EOR');
```
Comment (Mjwillis) [inline src/main/resources/db/dev/data/oasys/V10_3__role_data.sql]: Too much test data here, needs to be focused on just the application - i.e. what auth needs for a couple of users.

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/config/AuthenticationManagerConfiguration.java [hunk 7]
```diff
@@ -23,41 +23,47 @@
...
+    private final UserDetailsService oasysUserDetailsService;
     private final LoggingAccessDeniedHandler accessDeniedHandler;
     private final RedirectingLogoutSuccessHandler logoutSuccessHandler;
     private final JwtAuthenticationSuccessHandler jwtAuthenticationSuccessHandler;
     private final JwtCookieAuthenticationFilter jwtCookieAuthenticationFilter;
     private final String jwtCookieName;
     private final CookieRequestCache cookieRequestCache;
-    private final DaoAuthenticationProvider daoAuthenticationProvider;
+    private final DaoAuthenticationProvider nomisAuthenticationProvider;
+    private final DaoAuthenticationProvider oasysAuthenticationProvider;
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/digital/hmpps/oauth2server/config/AuthenticationManagerConfiguration.java]: Going to need some kind of toggle feature for Oasys until live.

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/config/AuthenticationManagerConfiguration.java [hunk 8]
```diff
@@ -23,41 +23,47 @@
...
+        this.nomisUserDetailsService = nomisUserDetailsService;
+        this.oasysUserDetailsService = oasysUserDetailsService;
         this.accessDeniedHandler = accessDeniedHandler;
         this.logoutSuccessHandler = logoutSuccessHandler;
         this.jwtAuthenticationSuccessHandler = jwtAuthenticationSuccessHandler;
         this.jwtCookieAuthenticationFilter = jwtCookieAuthenticationFilter;
         this.jwtCookieName = jwtCookieName;
         this.cookieRequestCache = cookieRequestCache;
-        this.daoAuthenticationProvider = daoAuthenticationProvider;
+        this.nomisAuthenticationProvider = nomisLockingAuthenticationProvider;
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/oauth2server/config/AuthenticationManagerConfiguration.java]: this authentication provider isn't just for nomis, but for auth.  That will get really confusing...

**Pull request closed without merging**